p2panda-sync/src/lib.rs

```diff
@@ -1,10 +1,44 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
-pub mod protocols;
-pub mod traits;
+#[cfg(feature = "cbor")]
+pub mod cbor;
+#[cfg(feature = "log-sync")]
+pub mod log_sync;
 
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use futures::{AsyncRead, AsyncWrite, Sink};
 use thiserror::Error;
 
+pub type TopicId = [u8; 32];
+
+/// Trait used for mapping a generic topic to a single or collection of logs
+pub trait TopicMap<K, V> {
+    fn get(&self, topic: &K) -> Option<V>;
+}
```

Comment:
> "TopicMap trait fn should be async and return a vec of log ids"

Ah, I think this was forgotten here

Reply:
Aha, true, I'll take that in a later PR.